packages/next/src/client/components/react-dev-overlay/ui/components/errors/dev-tools-indicator/dev-tools-info/segment-explorer.tsx

```diff
@@ -0,0 +1,199 @@
+      viewBox="0 0 16 16"
+      fill="none"
+      strokeLinejoin="round"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M14.5 6.5V13.5C14.5 14.8807 13.3807 16 12 16H4C2.61929 16 1.5 14.8807 1.5 13.5V1.5V0H3H8H9.08579C9.351 0 9.60536 0.105357 9.79289 0.292893L14.2071 4.70711C14.3946 4.89464 14.5 5.149 14.5 5.41421V6.5ZM13 6.5V13.5C13 14.0523 12.5523 14.5 12 14.5H4C3.44772 14.5 3 14.0523 3 13.5V1.5H8V5V6.5H9.5H13ZM9.5 2.12132V5H12.3787L9.5 2.12132Z"
+        fill="currentColor"
+      />
+    </svg>
+  )
+}
+
+const ICONS = {
+  layout: <IconLayout width={16} />,
+  page: <IconPage width={16} />,
+}
+
+function PageSegmentTree({ tree }: { tree: Trie | undefined }) {
+  if (!tree) {
+    return null
+  }
+  return (
+    <div className="segment-explorer-content">
+      <PageSegmentTreeLayerPresentation
+        tree={tree}
+        node={tree.getRoot()}
+        level={0}
+      />
+    </div>
+  )
+}
+
+function PageSegmentTreeLayerPresentation({
+  tree,
+  node,
+  level,
+}: {
+  tree: Trie
+  node: TrieNode
+  level: number
+}) {
+  const segments = node.value?.split('/') || []
+  const fileName = segments[segments.length - 1] || ''
+  const nodeName = fileName.split('.')[0] === 'layout' ? 'layout' : 'page'
+  const pagePath = segments.slice(0, -1).join('/')
+
+  return (
+    <div className="segment-explorer-item">
+      {!fileName || level === 0 ? null : (
+        <div className="segment-explorer-item-row">
+          <div className="segment-explorer-line">
+            <div className={`segment-explorer-line-text-${nodeName}`}>
+              <span
+                className={cx(
+                  'segment-explorer-line-icon',
+                  `segment-explorer-line-icon-${nodeName}`
+                )}
+              >
+                {nodeName === 'layout' ? ICONS.layout : ICONS.page}
+              </span>
+              {pagePath === '' ? '' : `${pagePath}/`}
+              <span className="segment-explorer-filename-path">{fileName}</span>
+            </div>
+          </div>
+        </div>
+      )}
+
+      <div className="tree-node-expanded-rendered-children">
+        {Object.entries(node.children).map(
+          ([key, child]) =>
+            child && (
+              <PageSegmentTreeLayerPresentation
+                key={key}
+                tree={tree}
+                node={child}
+                level={level + 1}
+              />
+            )
+        )}
+      </div>
+    </div>
+  )
+}
+
+export function SegmentsExplorer(
+  props: DevToolsInfoPropsCore & HTMLProps<HTMLDivElement>
+) {
+  const ctx = useSegmentTreeClientState()
+  if (!ctx) {
+    return null
+  }
+
+  // derive initial theme from system preference
+  return (
+    <DevToolsInfo
+      title={
+        <>
+          <button
+            className="segment-explorer-back-button"
+            onClick={props.close}
+          >
+            <LeftArrow />
```

Comment:
This isn't visible in dark mode

![CleanShot 2025-05-12 at 08 17 38@2x](https://github.com/user-attachments/assets/d4487e5a-d59f-44c9-9349-544be1e5a944)